### DISCUSSION TITLE
Missing functionality to clear value for country fields / select fields

### DISCUSSION THREAD

**Discussion by tpetry** [laravel/nova-issues#3213] [Q&A] [answered]
- Laravel Version: 8.25.0
- Nova Version: 3.20.0
- PHP Version: 8.0.0

### Description:
The current implementiation for the [country field](https://nova.laravel.com/docs/3.0/resources/fields.html#country-field) and [select field](https://nova.laravel.com/docs/3.0/resources/fields.html#select-field) has no ability to remove the value to set the value to `null`.
The only ability is to create fake values with no value anyone needs to select to remove the value.

### Detailed steps to reproduce the issue on a fresh Nova installation:
1. Add country or select field from the documentation to the fields
2. Update the resource by selecting one of the value and press save
3. Open the edit page again and try to remove the selected value to store a `null` value in the database

### How it is working currently:
Before you select a value
<img width="918" alt="Bildschirmfoto 2021-01-27 um 12 48 12" src="https://user-images.githubusercontent.com/315686/105987490-a4bed080-609e-11eb-8007-af2be156b956.png">

After you select a value:
<img width="931" alt="Bildschirmfoto 2021-01-27 um 12 48 04" src="https://user-images.githubusercontent.com/315686/105987531-af796580-609e-11eb-9dc3-446682283153.png">

If all selects would be transformed to the one which is searchable a clear button could be added next to the chevron down icon:
<img width="924" alt="Bildschirmfoto 2021-01-27 um 12 47 01" src="https://user-images.githubusercontent.com/315686/105987798-1434c000-609f-11eb-840e-3c4f4dbfea1a.png">

**Comment by crynobone** [comment, marked as answer]
> The only ability is to create fake values with no value anyone needs to select to remove the value.

If you set `nullable()` you should be able to use **Choose an option** to set as `null`.

**Reply by tpetry** [reply]
Selecting "Choose an option" is a nice trick to remove the value but this is far from usable! Try to teach non-technical people such workaround solutions, this will be very hard. For better UX there should be an option, as the new nova version will currently be developed someone should consider to solve this problem with the new nova.
